### PR TITLE
NCD - Labs

### PR DESCRIPTION
**Pull request by anvmn**
#416 
#417 
#419

### CODE REVIEW

**Comment by anvmn** [comment]
@mariano-dagostino 

Please review.

For the contect, we're making Labs activity with all it's subactivities (different types of lab tests) available globally, to be able to use at NCD module (so far it was only for Prenatal).
So, there's lots of renaming, code relocation and then appliance within NCD.

File: client/src/elm/Backend/Update.elm [hunk 1]
```diff
@@ -4792,25 +4846,159 @@ generatePrenatalLabsResultsAddedMsgs currentDate after test id =
...
+                                                let
+                                                    updatedCompletedTests =
+                                                        test :: completedTests
+
+                                                    resolutionDate =
+                                                        -- When all performed tests are completed, and Next Steps are either
+                                                        -- completed, or not required, setting today as resolution date.
+                                                        if
+                                                            (List.length updatedCompletedTests == List.length performedTests)
+                                                                && -- @todo
```
Comment (mariano-dagostino) [inline client/src/elm/Backend/Update.elm:4965]: @anvmn How do you track the pending `@todo` items? Perhaps creating the issue and linking it here?

**Review by mariano-dagostino — APPROVED** [review]
@anvmn Looks good. Just a question about how you handle the 5 or so `@todo` items you added in this PR?

File: client/src/elm/Backend/Update.elm (hunk 1 above)
Comment (anvmn) [reply client/src/elm/Backend/Update.elm:4965]: @mariano-dagostino 

When developing a module which 'replicates' the infrastructure of similar module (for example, additional type of individual encounter), there're quire alot of @todos that are not related to current PR development and are completed at follow up PRs.
Before submiting PR for review, I verify none of @todos are needed in this PR.
Once module is competed, I verify there are no @todos left.

**Pull request merged**